tests/convergence/ex1cp.cpp

```diff
@@ -0,0 +1,449 @@
+   }
+   else if (pmesh->GetNodes())
+   {
+      fec = pmesh->GetNodes()->OwnFEC();
+      if (myid == 0)
+      {
+         cout << "Using isoparametric FEs: " << fec->Name() << endl;
+      }
+   }
+   else
+   {
+      fec = new H1_FECollection(order = 1, dim);
+   }
+   ParFiniteElementSpace *fespace = new ParFiniteElementSpace(pmesh, fec);
+   HYPRE_Int size = fespace->GlobalTrueVSize();
+   if (myid == 0)
+   {
+      cout << "Number of finite element unknowns: " << size << endl;
+   }
+
+   prob_kind = (prob)iprob;
+   mfem::ComplexOperator::Convention conv = mfem::ComplexOperator::HERMITIAN;
+
+   // Set up the parallel linear form b(.)
+   ParComplexLinearForm *b = new ParComplexLinearForm(fespace,conv);
+   FunctionCoefficient f_re(f_exact_re);
+   FunctionCoefficient f_im(f_exact_im);
+   b->AddDomainIntegrator(new DomainLFIntegrator(f_re),
+                          new DomainLFIntegrator(f_im));
+   b->real().Vector::operator=(0.0);
+   b->imag().Vector::operator=(0.0);
+
+   // Define the solution vector x as a parallel finite element grid function
+   // corresponding to fespace
+   ParComplexGridFunction x(fespace);
+   ParComplexGridFunction x_ex(fespace);
+   FunctionCoefficient u_ex_re(u_exact_re);
+   FunctionCoefficient u_ex_im(u_exact_im);
+   VectorFunctionCoefficient u_grad_coeff_re(dim, u_grad_exact_re);
+   VectorFunctionCoefficient u_grad_coeff_im(dim, u_grad_exact_im);
+
+   // Set up the parallel bilinear form a(.,.)
+   ConstantCoefficient one(1.0);
+   ParSesquilinearForm *a = new ParSesquilinearForm(fespace,conv);
+   switch (prob_kind)
+   {
+      case comp :
+         a->AddDomainIntegrator(new DiffusionIntegrator(one),
+                                new DiffusionIntegrator(one));
+         break;
+      case rl :
+         a->AddDomainIntegrator(new DiffusionIntegrator(one), NULL);
+         break;
+      case im :
+         a->AddDomainIntegrator(NULL,new DiffusionIntegrator(one));
+         break;
+   }
+
+   double L2err_re0, L2err_im0;
+   double H1err_re0, H1err_im0;
+
+   for (int l=0; l<ref_levels+1; l++)
+   {
+      a->Assemble();
+      b->Assemble();
+      OperatorHandle Ah;
+      Vector B, X;
+      // Determine the list of true (i.e. conforming) essential boundary dofs.
+      Array<int> ess_bdr, ess_tdof_list;
+      if (pmesh->bdr_attributes.Size())
+      {
+         ess_bdr.SetSize(pmesh->bdr_attributes.Max());
+         ess_bdr = 1;
+         fespace->GetEssentialTrueDofs(ess_bdr, ess_tdof_list);
+      }
+      x.ProjectBdrCoefficient(u_ex_re,u_ex_im,ess_bdr);
+      x_ex.ProjectCoefficient(u_ex_re,u_ex_im);
+
+      a->FormLinearSystem(ess_tdof_list, x, *b, Ah, X, B);
+
+      ComplexHypreParMatrix * AZ = Ah.As<ComplexHypreParMatrix>();
+      HypreParMatrix * A = AZ->GetSystemMatrix();
+
+      if (myid == 0)
+      {
+         cout << "Size of linear system: " << A->GetGlobalNumRows() << endl;
+      }
+
+#ifndef MFEM_USE_SUPERLU
+      {
+         GMRESSolver gmres(MPI_COMM_WORLD);
```

Comment:
This has convergence issues (not surprisingly), particularly at higher orders.  Perhaps we should try using AMG in a block diagonal preconditioner.

Reply:
Fixed and works fine now